### PR TITLE
Kl mr new list

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items
## Summary 
A shopping list is a set of items associated with a specific three-word token. We need to allow users to start new lists, so they can save the things they need to buy. To start a new list, we will need to do a few things:

1. Generate a new, unique token
2. Save the token to `localStorage`
3. Show the user the list view

Up to this point, the app has rendered the items in the list associated with the token `my test list`. Before we can allow users to make *new* lists, we have to stop showing them this test list.

## Acceptance criteria
- [x] `@the-collab-lab/shopping-list-utils` is added as a dependency to the project.
- [x] The string `my test list` in `App.jsx` is replaced with `null`, so users are not automatically subscribed to any list.

If a user **doesn’t** already have a token:
- [x] A button in the `Home` component allows them to create a new list
- [x] Clicking the button generates a new token and saves it to `localStorage` using the `setListToken` function in `App.jsx`.
- [x] Once the token has been created and saved, the user is redirected to the `List` view.

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- The  `generateToken` function from the `shopping-list-utils` package should be used to generate the token. More information on that package [its documentation](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils). Importing it will look something like this:
``` js
import { generateToken } from '@the-collab-lab/shopping-list-utils';
```
- The `setListToken` function takes **one** argument: the new list token.
- Eventually the welcome/home page will look like [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png). It’s good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
- For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by kaitlinLewis**
## Description

This allows the creation of unique 3 word tokens that enables users to save specific items for their shopping list. In order to accomplish this, we used local storage so that each user can have their own list as local storage is unique to each browser. It does not have an expiry date for the information stored in local storage so this will allow users to return to that same list. In order to redirect users to the List page after the creation of a unique token, the useNavigate hook was used. This was helpful as it allows for programmatic navigation. If the code was to make use of Classes, it would be straightforward to replace the useNavigate with the Navigate component instead.
 
## Related Issue

Closes #3 

## Acceptance Criteria

@the-collab-lab/shopping-list-utils is added as a dependency to the project.
 
The string my test list in App.jsx is replaced with null, so users are not automatically subscribed to any list.

If a user doesn’t already have a token:
A button in the Home component allows them to create a new list
Clicking the button generates a new token and saves it to localStorage using the setListToken function in App.jsx.
Once the token has been created and saved, the user is redirected to the List view.

If a user does already have a token:
They are automatically redirected to the List view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

Screen Shot 2022-10-13 at 6 59 40 PM

### After

Screen Shot 2022-10-13 at 7 00 21 PM

## Testing Steps / QA Criteria

When the "Create new list" button is clicked, there is a redirect to List page which is empty.
In the inspect tools, in local storage, there is a 3 word token.

### CODE REVIEW

**Comment by NuriyaAkh** [comment]
nice job!

**Review by NuriyaAkh — APPROVED** [review]

**Review by snowsneakers — APPROVED** [review]
Looks great

File: src/App.jsx [hunk 1]
```diff
@@ -48,14 +59,14 @@ export function App() {
...
-		</Router>
+		//<Router>
+		<Routes>
+			<Route path="/" element={<Layout />}>
+				<Route index element={<Home handleClick={handleClick} />} />
+				<Route path="/list" element={<List data={data} />} />
+				<Route path="/add-item" element={<AddItem />} />
+			</Route>
+		</Routes>
+		//</Router>
```
Comment (cafloyd) [inline src/App.jsx:70]: Let's go ahead and remove the `<Router/>` here entirely instead of commenting it out, since you've moved it to `index.jsx`!

**Review by cafloyd — APPROVED** [review]
Fantastic work! Great job figuring out how to work with the router and local storage! Also very happy to see y'all talking through your process in the PR description - extremely helpful for your reviewers!! 🤩

Nice job too thinking about the big picture and how to keep our code base nice and clean with your changes. One tiny suggestion but otherwise looks ready to merge to me!

**Pull request merged**